src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -0,0 +1,133 @@
+import { URI } from 'vs/base/common/uri';
+import * as path from 'vs/base/common/path';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { replaceInvalidLinkPath } from 'sql/workbench/contrib/notebook/common/utils';
+
+const keepAbsolutePathConfigName = 'notebook.keepAbsolutePath';
+
+export class NotebookLinkHandler {
+	private _notebookUriLink: URI | undefined;
+	private _href: string | undefined;
+	private _notebookDirectory: string;
+	private _isAnchorLink: boolean | undefined;
+	private _isFile: boolean | undefined;
+	private _isAbsolutePath: boolean | undefined;
```

Comment:
All these boolean values don't look like they can actually be undefined since you assign them in the constructor.